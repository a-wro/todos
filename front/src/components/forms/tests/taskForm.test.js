import React from 'react'
import { shallow } from 'enzyme'
import TaskForm from '../taskForm'

describe('TaskForm', () => {
    const testData = 'test'
    const mockChange = jest.fn()
    const mockSubmit = jest.fn()

    const props = {
        handleChange: mockChange,
        handleSubmit: mockSubmit,
    }
    
    const taskForm = shallow(<TaskForm {...props} />)
    
    describe('rendering', () => {
        it('renders correctly', () => {
            expect(taskForm).toMatchSnapshot()
        })
    })

    describe('handling input', () => {
        const event = { target: { value: testData }}
        beforeEach(() => {
            taskForm.find('.task-input').simulate('change', event)
        })

        it('onChange handler called', () => {
            expect(mockChange).toHaveBeenCalledWith(event)
        })

        it('local state updates', () => {
            expect(taskForm.state().term).toEqual(testData)
        })
    })

    describe('submitting input', () => {
        beforeEach(() => {
            taskForm.find('.task-form').simulate('submit', {
                preventDefault: () => {},
            })
        })

        it('submits properly', () => {
            expect(mockSubmit).toHaveBeenCalled()
        })
    
    
    })
       
    
    

    

})