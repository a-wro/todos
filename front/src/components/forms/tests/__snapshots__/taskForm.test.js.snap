// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm rendering renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskForm
    handleChange={[Function]}
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="task-input"
          onChange={[Function]}
          placeholder="Enter a task"
          type="text"
          value=""
/>,
        <input
          type="submit"
          value="add task"
/>,
      ],
      "className": "task-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "task-input",
          "onChange": [Function],
          "placeholder": "Enter a task",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "submit",
          "value": "add task",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="task-input"
            onChange={[Function]}
            placeholder="Enter a task"
            type="text"
            value=""
/>,
          <input
            type="submit"
            value="add task"
/>,
        ],
        "className": "task-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "task-input",
            "onChange": [Function],
            "placeholder": "Enter a task",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
            "value": "add task",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
